connect/runtime/src/test/java/org/apache/kafka/connect/integration/TaskHandle.java

```diff
@@ -37,22 +39,26 @@
     private final ConnectorHandle connectorHandle;
     private final AtomicInteger partitionsAssigned = new AtomicInteger(0);
     private final StartAndStopCounter startAndStopCounter = new StartAndStopCounter();
+    private final Consumer<SinkRecord> consumer;
 
     private CountDownLatch recordsRemainingLatch;
     private CountDownLatch recordsToCommitLatch;
     private int expectedRecords = -1;
     private int expectedCommits = -1;
 
-    public TaskHandle(ConnectorHandle connectorHandle, String taskId) {
-        log.info("Created task {} for connector {}", taskId, connectorHandle);
+    public TaskHandle(ConnectorHandle connectorHandle, String taskId, Consumer<SinkRecord> consumer) {
         this.taskId = taskId;
         this.connectorHandle = connectorHandle;
+        this.consumer = consumer;
     }
 
     /**
      * Record a message arrival at the task and the connector overall.
      */
-    public void record() {
```

Comment:
This might break existing tests that depend on Connect's integration tests framework. Probably good idea to keep it, in which case the consumer should be ignored.